.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,96 @@
+name: Run integration tests
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  push:
+    branches:
+      - master
+      - v*
+  pull_request:
+    branches:
+    - master
+    - v*
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.after }}
+  cancel-in-progress: true
+
+env:
+  HELM_VERSION: v3.10.3
+  KIND_VERSION: v0.17.0
+  KIND_CONFIG: .github/kind-config.yaml
+  CILIUM_VERSION: 1.12.5
+  CILIUM_VALUES: .github/cilium-values.yaml
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+      - name: Setup go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568
+        with:
+          go-version: '1.19.5'
+      - name: Build hubble CLI
+        run: make
+      - name: Set up Helm
+        uses: azure/setup-helm@5119fcb9089d432beecbf79bb2c7915207344b78
+        with:
+          version: ${{ env.HELM_VERSION }}
+      - name: Create kind cluster
+        uses: helm/kind-action@d8ccf8fb623ce1bb360ae2f45f323d9d5c5e9f00
+        with:
+          version: ${{ env.KIND_VERSION }}
+          config: ${{ env.KIND_CONFIG }}
+          cluster_name: kind
+      - name: Deploy Cilium
+        run: |
+          # Deploy cilium with KPR
+          MASTER_IP="$(docker inspect kind-control-plane | jq '.[0].NetworkSettings.Networks.kind.IPAddress' -r)"
+          helm repo add cilium https://helm.cilium.io
+          helm repo update
+          helm install cilium cilium/cilium \
+            --wait \
+            --namespace kube-system \
+            --version ${{ env.CILIUM_VERSION }} \
+            --values ${{ env.CILIUM_VALUES }} \
+            --set kubeProxyReplacement=strict \
+            --set k8sServiceHost="${MASTER_IP}" \
+            --set k8sServicePort=6443
+      - name: Wait for hubble-relay to be running
+        run: |
+          kubectl -n kube-system rollout status deployment/hubble-relay
+      - name: Run integration tests
+        run: |
+          set -ex
+          ./hubble --version
+
+          kubectl -n kube-system port-forward service/hubble-relay 4245:80 &
+          sleep 5s
+          test $(pgrep --count --full "kubectl.*port-forward.*service\/hubble-relay.*4245:80") -eq 1
+
+          # query hubble
+          ./hubble observe -n kube-system
```

Comment:
We can go arbitrarily far with this - I do wonder however if we should check that there is at least some output here, since Hubble will return exit code 0 even if there are no flows.

Reply:
How about something like

```
./hubble observe -n kube-system -o jsonpb | tee flows.json
test $(jq --slurp length flows.json) -gt 0
```

From there we can re-use `flows.json`, e.g. to test pipe to the CLI

```
test $(./hubble observe < flows.json) -eq $(wc -l flows.json)
```

Reply:
Yeah I think eventually I'd like to actually move these into Go based tests we run, or even just [bats](https://github.com/bats-core/bats-core/). I can definitely make it check for flows though, that's a good idea.